cpp/src/arrow/compare.cc

```diff
@@ -544,8 +554,8 @@ class ArrayEqualsVisitor : public RangeEqualsVisitor {
       }
     } else {
       // ARROW-537: Only compare data in non-null slots
-      const int32_t* left_offsets = left.raw_value_offsets();
-      const int32_t* right_offsets = right.raw_value_offsets();
+      const auto left_offsets = left.raw_value_offsets();
```

Comment:
clarity note: `const auto left_offsets` is equivalent to `const offset_type* const left_offsets` here

Reply:
```suggestion
      auto left_offsets = left.raw_value_offsets();
```

Reply:
Ah, thanks. Indeed, removing the const can clarify.